### ISSUE TITLE
getInitialNotification is called on every dev "Reload" after opening a notification

### ISSUE THREAD

**Issue by BatDroid** [evollu/react-native-fcm#1020]
react-native : 55.1
react-native-fcm: 16.1.0

whenever i open the app from notification (wether the app had been in background or not Running) and then reloading the app in dev mode the `getInitialNotification` is called and this brings bugs to me in dev mode since based on the notification i navigate user to a screen.(by react-native-router-flux) after opening the notification whenever i press Reload (to refresh the app) `getInitialNotification` is called with the same notification object. i've tried `removeAllDeliveredNotification` but it didn't fix the problem. it is in both ios and android

**Comment by evollu** [comment]
where do you call getInitialNotification?

**Comment by BatDroid** [comment]
@evollu at index.android.js/index.ios.js before outside the component. but i think this is related to the first bullet point of [here](https://github.com/evollu/react-native-fcm#behaviour-when-sending-notification-and-data-payload-through-gcm). i wish there was a way to fix this or ill have to keep the notifications in a database.

**Comment by evollu** [comment]
so the problem only happens when you open the app though notification and hit cmd+R to reload JS?
sounds a rare use case to me and shouldn't happen in production because you can't reload the app

**Comment by BatDroid** [comment]
yes, but unfortunately it happens in my app. my app is the type which user usually opens it when receives push notification and i have implemented codepush in my app. so the scenario (which is not rare) is user opens the app via push notification and then receives a codepush Update and then the js reloads and now there is a problem!

**Comment by evollu** [comment]
I can't think of any way other than tracking it in storage.

**Comment by BatDroid** [comment]
- i think thats the only solution without changing anything here
- i will try but i don't think that work since the js is reloaded completely in codepush and getting the current route gives the entry route always
- yes that's inside of the componentDidMount of index.android/ios.js
